src/main/java/frc/utils/math/FieldMath.java

```diff
@@ -0,0 +1,84 @@
+package frc.utils.math;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.geometry.Translation3d;
+
+import frc.constants.MathConstants;
+import frc.constants.field.Field;
+
+public class FieldMath {
+
+	public static Translation2d getRelativeTranslation(Translation2d relativeTo, Translation2d toRelative) {
+		return toRelative.minus(relativeTo);
+	}
+
+	public static Translation2d getRelativeTranslation(Pose2d relativeTo, Translation2d toRelative) {
+		return getRelativeTranslation(relativeTo.getTranslation(), toRelative).rotateBy(relativeTo.getRotation().unaryMinus());
+	}
+
+	public static Rotation2d getMirroredAngle(Rotation2d angle) {
+		return MathConstants.HALF_CIRCLE.minus(angle);
+	}
+
+	public static Rotation2d getInvertedAngle(Rotation2d angle) {
+		return MathConstants.HALF_CIRCLE.plus(angle);
+	}
+
+	public static double getMirroredX(double x) {
+		return Field.LENGTH_METERS - x;
+	}
+
+	public static double getMirroredY(double y) {
+		return Field.WIDTH_METERS - y;
+	}
+
+	public static Pose2d getMirrored(Pose2d pose2d, boolean mirrorY, boolean mirrorX, boolean invertAngle, boolean mirrorAngle) {
```

Comment:
Not critical but maybe make this a record or something? this is a lot of arguments.
like
```suggestion
	public static Pose2d getMirrored(Pose2d pose2d, MirrorSettings mirrorSettings) {
```

Reply:
just for this thing i dont think need, maybe in the future